sensor/common/imageutil/util.go

```diff
@@ -8,10 +8,18 @@ import (
 // IsInternalImage determines if the image represented by the given name
 // is an "internal" image. An internal image is one which is hosted by an internal registry.
 // An internal registry is on which is only accessible from within the cluster in which it lives.
-func IsInternalImage(image *storage.ImageName) bool {
+// IsInternalImage uses the data from the given store to determine this.
+// If registryStore is nil, registry.Singleton() is used as the store.
+func IsInternalImage(image *storage.ImageName, registryStore *registry.Store) bool {
 	// If the Sensor knows about the registry in which the image is hosted,
 	// then the image must be "internal" to the cluster, as Sensor only tracks
 	// "internal" registries.
-	reg, err := registry.Singleton().GetRegistryForImage(image)
+	var store *registry.Store
```

Comment:
I would modify this and instead just ensure that registryStore is always set to a valid value?

Reply:
I think having a pure function that then has a Singleton inside of it is a weird side effect of the function if you pass the wrong parameters. I agree that `(*registry.Store).HasRegistryForImage` would be cleaner